Documentation for LocalQueueDefaulting (#3783)
kubernetes-sigs/kueue: site/content/en/docs/tasks/manage/enforce_job_management/manage_jobs_without_queue_name.md

---

> title: "Manage Jobs without queue name"

Reviewer: Actually, I would suggest moving the content about defaulting LocalQueues out to a separate subpage. Sibling to this one. The reason being that the defaulting is also relevant for VAP, so there is no reason to couple it to manageJobsWithoutQueueName . The title could be "Setup LocalQueue defaulting".

Author: I think currently default local queue feature related to how manage jobs without queue name and it would make sense to have all settings in one place, so users will understand how those feature work together and what the difference between them. 

However what is VAP? Maybe it worth to add the link to default LocalQueue section in other places.

Reviewer: https://github.com/kubernetes-sigs/kueue/blob/main/site/content/en/docs/tasks/manage/enforce_job_management/setup_job_admission_policy.md

Reviewer: > I think currently default local queue feature related to how manage jobs without queue name and it would 
> make sense to have all settings in one place, so users will understand how those feature work together and 
> what the difference between them.

I would rather link the dedicated page from both subpages for manageJobsWithoutQueueName and VAP. Since the defaulting happens in webhook it applies to both features. "have all settings in one place" - yeah in the recent update of docs on main the three related features would be under the common folder `enforce_job_management`.

Reviewer: For better visibility of the feature we may also link it from https://kueue.sigs.k8s.io/docs/reference/labels-and-annotations/#kueuex-k8sioqueue-name

Author: Done

Learn how to [install Kueue with a custom manager configuration](/docs/installation/#install-a-custom-configured-released-version).

## Setup default LocalQueue

> LocalQueueDefaulting is an Alpha feature gate (disabled by default) that allows to use LocalQueue with `default` name as a default LocalQueue for the workloads in the same namespace without queue `kueue.x-k8s.io/queue-name` label. To use a feature:

Reviewer 2: ```suggestion
LocalQueueDefaulting is an Alpha feature that allows the use of a LocalQueue with name `default` as the default LocalQueue for workloads in the same namespace that do not have the `kueue.x-k8s.io/queue-name` label. The feature is gated by a feature the `LocalQueueDefaulting` feature gate, and is disabled by default. To use this feature:
```

> LocalQueueDefaulting is an Alpha feature that allows the use of a LocalQueue with name `default` as the default LocalQueue for workloads in the same namespace that do not have the `kueue.x-k8s.io/queue-name` label. The feature is gated by a feature the `LocalQueueDefaulting` feature gate, and is disabled by default. To use this feature:

Reviewer 2: ```suggestion
LocalQueueDefaulting is an Alpha feature that allows the use of a LocalQueue with name `default` as the default LocalQueue for workloads in the same namespace that do not have the `kueue.x-k8s.io/queue-name` label. The feature is gated by the `LocalQueueDefaulting` feature gate, and is disabled by default. To use this feature:
```

> - Enable LocalQueueDefaulting feature gate. Check the [Installation](/docs/installation/#change-the-feature-gates-configuration) guide for details on feature gate configuration.

Reviewer 2: ```suggestion
- Enable the LocalQueueDefaulting feature gate. Refer to the [feature gates configuration](/docs/installation/#change-the-feature-gates-configuration) guide for details.
```

> - create LocalQueue with the `default` name.

Reviewer 2: ```suggestion
- create a LocalQueue with the name `default`.
```

- create workload in the same namespace and observe that workload is updated with `kueue.x-k8s.io/queue-name:default` label.

> - workloads that created in different namespace or workloads that has `kueue.x-k8s.io/queue-name` label won't be modified.

Reviewer 2: ```suggestion
- workloads created in a different namespace or workloads that already have the `kueue.x-k8s.io/queue-name` label won't be modified.
```
